code/controllers/subsystem/xenoarchaeology.dm

```diff
@@ -0,0 +1,241 @@
+SUBSYSTEM_DEF(xenoarchaeology)
+	name = "Xenoarchaeology"
+	flags = SS_NO_FIRE
+	init_order = INIT_ORDER_XENOARCHAEOLOGY
+
+	///Which console is the main character
+	var/obj/machinery/computer/xenoarchaeology_console/main_console
+
+	//All the traits - before we sort them, semi needed for generation
+	var/list/xenoa_all_traits
+	///All traits indexed by name, used for labelling stuff
+	var/list/xenoa_all_traits_keyed
+
+	///Names for science sellers & artifacts
+	var/list/xenoa_seller_names
+	var/list/xenoa_seller_dialogue
+	var/list/xenoa_artifact_names
+
+	///Whitelist for traits by material type
+	var/list/material_traits
+
+	///Incompatability lists - Partial future proofing some stuff
+	var/list/xenoa_item_incompatible
+	var/list/xenoa_mob_incompatible
+	var/list/xenoa_structure_incompatible
+
+	///Labeler trait lists, basically just names
+	var/datum/xenoa_material_traits/stats/labeler_traits
+	///Other labeler shit
+	var/list/labeler_tooltip_stats = list()
+	var/list/labeler_traits_filter
+
+	///Material weights, basically rarity - Also used to populate some other lists
+	var/list/xenoartifact_material_weights = list(XENOA_BLUESPACE = 10, XENOA_PLASMA = 8, XENOA_URANIUM = 5, XENOA_BANANIUM = 1)
+
+	///Trait priority list - The order is important and it represents priotity
+	var/list/xenoartifact_trait_priorities = list(TRAIT_PRIORITY_ACTIVATOR, TRAIT_PRIORITY_MINOR, TRAIT_PRIORITY_MALFUNCTION, TRAIT_PRIORITY_MAJOR)
+
+	///List of 'discovered' traits
+	var/list/discovered_traits = list()
+
+/datum/controller/subsystem/xenoarchaeology/Initialize(timeofday)
+	. = ..()
+	//Poplate seller & artifact personalities
+	xenoa_seller_names = world.file2list("strings/names/science_seller.txt")
+	xenoa_seller_dialogue = world.file2list("strings/science_dialogue.txt")
+	xenoa_artifact_names = world.file2list("strings/names/artifact_sentience.txt")
```

Comment:
in a rare case where that the game failed to get these texts, there should be a default one that can cover. I guess some IC-friendly error message?